dndsim/src/classes/Barbarian.ts

```diff
@@ -0,0 +1,254 @@
+    }
+
+    damageRoll(event: DamageRollEvent) {
+        if (event.attack?.attack?.weapon()?.mod(this.character) == "str") {
+            event.damage.flatDmg += this.character.getAttribute(RageBonusDamageAttribute);
+        }
+    }
+
+    shortRest() {
+        this.character.removeEffect(RageEffect)
+    }
+
+    addResource() {
+        this.character.resources.set(
+            RageResource,
+            new Resource({
+                name: RageResource,
+                character: this.character,
+                initialMax: 2,
+                incrementOnShortRest: true,
+                resetOnLongRest: true,
+
+            })
+        )
+    }
+}
+
+class AddRage extends Feat {
+    apply(character: Character): void {
+        character.getResource(RageResource).addMax(1)
+    }
+}
+
+class RecklessAttack extends Feat {
+    apply(character: Character) {
+        character.events.on("attack_roll", (event) => this.attackRoll(character, event))
+    }
+
+    attackRoll(character: Character, event: AttackRollEvent) {
+        if (event.attack?.attack?.weapon()?.mod(character) == "str") {
+            event.adv = true
+            event.addTag(RecklessTag)
+        }
+    }
+}
+
+class PrimalChampion extends Feat {
+    apply(character: Character) {
+        character.increaseStatAndMax("str", 4)
+        character.increaseStatAndMax("con", 4)
+    }
+}
+
+class Frenzy extends Feat {
+    applied = false
+
+    apply(character: Character) {
+        character.events.on("begin_turn", () => this.applied = false)
+        character.events.on("damage_roll", (event) => this.damageRoll(event))
+    }
+
+    damageRoll(event: DamageRollEvent) {
+        if (!this.applied && event.attack?.hasTag(RecklessTag)) {
+            const dice = Array(this.character.getAttribute(RageBonusDamageAttribute)).fill(6)
+            event.damage.addDice(dice)
+            this.applied = true
+        }
+    }
+}
+
+class DivineFury extends Feat {
+    applied = false
+
+    apply(character: Character) {
+        character.events.on("begin_turn", () => this.applied = false)
+        character.events.on("damage_roll", (event) => this.damageRoll(event))
+    }
+
+    damageRoll(event: DamageRollEvent) {
+        if (!this.applied && event.attack?.attack instanceof WeaponAttack) {
+            event.damage.addDice([6])
+            event.damage.flatDmg += Math.floor(this.character.getClassLevel("Barbarian") / 2)
+            this.applied = true
+        }
+    }
+}
+
+export class Barbarian {
+    static baseFeats(args: {
+        level: number
+        asis: Array<Feat>
+        masteries: WeaponMastery[]
+    }): Feat[] {
+        const { level, asis, masteries } = args
+        const feats: Feat[] = []
+        if (level >= 1) {
+            feats.push(new ClassLevel("Barbarian", level))
+            feats.push(new WeaponMasteries(masteries))
+            feats.push(new SetAttribute(RageBonusDamageAttribute, 2));
```

Comment:
I added `SetAttribute` and `IncreaseResource` as their own feats since they should be pretty common.

Reply:
I wonder if it would be cleaner to inline the entire thing into the `Rage` feat and have it decide both the rage bonus and the number of resource usages based on the barbarian level, instead of making a different feat for each level where anything was incremented.

Reply:
I'll go with the latter for now for consistency.